### ISSUE TITLE
Changing owner of group does not remove original owner

### ISSUE THREAD

**Issue by jwhiteman36** [RocketChat/Rocket.Chat#15034]
**Description:**

Changing owner of a room does not remove the original owner. The channel posts that the owner was changed but the user is still the owner if you pull the channel info using the API. The user can also still be removed as owner using the API, the channel posts this change as well. If you pull the channel info again after removing the owner, it still shows them as the owner of the channel. I am unable to delete users because of this. 

**Steps to reproduce:**

	1. Create new room with another user
	2. Change owner to another user
	3. Attempt to delete the user that created the room
	4. Receive error that user is the owner of a room

**Expected behavior:** 

Original room owner should be removed as owner when another user is assigned owner. 

**Actual behavior:**

Both users are being assigned ownership of room. Now unable to delete the user that was the original owner.

**Server setup:**

	- Version of Rocket.Chat Server: 1.2.1
	- Operating System: CentOS 7
	- Deployment Method: Manual
	- Number of running instances: 10
	- DB Replicaset Oplog:
	- NodeJS Version: 8.11.4
        - MongoDB Version: 3.6.11

**Comment by Beeez** [comment]
I am having this issue too.

**Comment by ggazzo** [comment]
I think the problem its a little bit different, we can have more than 1 owner, the problem is when we are deleting the room, we are not identifying that we have another owner.

**Comment by ggazzo** [comment]
@jwhiteman36 actually I made 6 tests and I couldnt reproduce your problem, please check your steps and let me know if I'm doing something wrong

## test 1
- create a user `g2`
- as `g2` create a `public` room `test`
- as `g2` invite user `g1` to `test`
- as `g2` select user `g1` on users room
- set `g1` as owner
- as a `g1` (admin) go to Admin -> users -> `g2` -> delete user
- success


## test 2
- create a user `g2`
- as `g2` create a `private` room `test`
- as `g2` invite user `g1` to `test`
- as `g2` select user `g1` on users room
- as `g2` set `g1` as owner
- as a `g1` (admin) go to Admin -> users -> `g2` -> delete user
- success

## test 3
- create a user `g2`
- as `g2` create a `direct message` room  with `g1`
- as `g2` invite user `g1` to `test`
- as `g2` select user `g1` on users room
- as `g2` set `g1` as owner
- as `g1` (admin) go to Admin -> users -> `g2` -> delete user
- success

## test 4
- create a user `g2`
- as `g2` create a `private` room `test`
- as `g2` invite user `g1` to `test`
- as `g2` select user `g1` on users room
- as `g2` set `g1` as owner
- as a `g2` go to Profile -> Delete my account
- success

## test 5
- create a user `g2`
- as `g2` create a `public` room `test`
- as `g2` invite user `g1` to `test`
- as `g2` select user `g1` on users room
- as `g2` set `g1` as owner
- as a `g2` go to Profile -> Delete my account
- success

## test 6
- create a user `g2`
- as `g2` create a `direct message` room  with `g1`
- as `g2` invite user `g1` to `test`
- as `g2` select user `g1` on users room
- as `g2` set `g1` as owner
- as a `g2` go to Profile -> Delete my account
- success

**Comment by jwhiteman36** [comment]
I attempted to reproduce this issue again by creating a new user and group to test off of, I also was able to delete that new user created for the test. I found that the only users I am not able to delete after setting a new owner of a group are those that were present before we updated from version 1.1.3 to 1.2.1. The channel does reflect the change in the owner, but when I pull the group information using the API the owner still shows as the original.
